programs/serializers.py

```diff
@@ -41,12 +55,31 @@ class Meta:
     def get_programs(self, obj: ProgramCategory):
         return ProgramSerializer(obj.programs.filter(active=True), many=True).data
 
+    def get_icon(self, obj: ProgramCategory):
+        return obj.icon.name if obj.icon else "default"
+
 
 class UrgentNeedAPISerializer(serializers.ModelSerializer):
     name = ModelTranslationSerializer()
     website_description = ModelTranslationSerializer()
-    type = ModelTranslationSerializer()
 
     class Meta:
         model = UrgentNeed
-        fields = ("id", "name", "website_description", "type")
+        fields = ("id", "name", "website_description")
+
+
+class UrgentNeedTypeSerializer(serializers.ModelSerializer):
+    name = ModelTranslationSerializer()
+    urgent_needs = serializers.SerializerMethodField()
+    icon = serializers.SerializerMethodField()
```

Comment:
I think that you can do this because `icon_name` is a property.
```suggestion
    icon = serializers.CharField(source="icon_name")
```